### PR TITLE
Create a ReadTheDocs page and demo for DYAD

### PR DESCRIPTION
**Pull request by ilumsden**
This PR adds the content for a Sphinx webpage (which can then be pushed to ReadTheDocs). This webpage contains the following pages:
* `index.rst`: basic intro to DYAD plus its benefits
* `getting_started.rst`: information about building, installing, and using DYAD
* `ecp_feb_2023_tutorial.rst`: currently just a placeholder, but will eventually contain information about the upcoming DYAD demo in the Flux ECP tutorial
* `publications.rst`: a list of citations to all the publications associated with DYAD (based on a similar page in the [Hatchet documentation](https://llnl-hatchet.readthedocs.io/en/latest/publications.html))
* `developer_guide.rst`: information about contributing, developer guidelines, etc. This really just amounts to pointing to the equivalent information from Flux.

All of this can be found in the `docs` subdirectory.

Additionally, this PR adds a demo for the upcoming ECP tutorial. The demo can be found in `docs/demos/ecp_feb_2023`. This demo can later be repurposed for regression testing.

### CODE REVIEW

File: docs/bib/posters.bib [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@INPROCEEDINGS{escience2022-poster,
+    author={Yeom, Jae-Seung and Ahn, Dong H. and Lumsden, Ian and Luettgau, Jakob and Caino-Lores, Silvina and Taufer, Michela},
```
Comment (JaeseungYeom) [inline docs/bib/posters.bib:2]: You might have to use `Jae{-}Seung` to make sure it appears correctly.

**Review by JaeseungYeom — APPROVED** [review]
Looks good.
However, the c-code needs to follow flux coding style.

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   sphinx-quickstart on Thu Jan 26 10:12:01 2023.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+DYAD: Dynamic and Asynchronous Data Streamliner
+===============================================
+
+DYAD is a synchronization and data movement tool for computational science workflows built
+on top of Flux. DYAD aims to provide the benefits of in situ and in transit tools
+(e.g., fine-grained synchronization of consumer applications, fast data access due to spatial locality)
```
Comment (JaeseungYeom) [inline docs/index.rst:11]: `of consumer applications` It is the synchronization between producer and consumer, or (read and write)

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+* Lack of per-file or per-data object synchronization in shared-storage approaches
+* Poor temporal and spatial locality in shared-storage approaches
+* Poor performance for file/data object metadata operations in shared-storage approaches (and possibly some in situ and in transit approaches)
+* Poor portability and the introduction of required code changes for in situ and in transit approaches
+
+In resolving these challenges, DYAD aims to provide the following to users:
+
+* Good performance (similar to in situ and in transit) due to on- or near-node temporary storage of data
+* Per-file or per-data object synchronization of consumer applications
```
Comment (JaeseungYeom) [inline docs/index.rst]: Lets remove per-data object for now. We are not there yet.

File: docs/getting_started.rst [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+The namespace can be whatever string value you want.
+
+Determine the Managed Directories for Each Application
+******************************************************
+
+To determine when to perform synchronization and data transfer, DYAD tracks two directories for
+each application: a **producer-managed directory** and a **consumer-managed directory**. At least
+one of these directories must be specified for DYAD to do anything. If neither are provided, the application
+will still run, but DYAD will not do anything.
```
Comment (JaeseungYeom) [inline docs/getting_started.rst:201]: Mention that DYAD will not interfere the I/O operation on files under a path that is not managed.

File: docs/getting_started.rst [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+
+1. :ref:`Create a Flux key-value store (KVS) namespace <Create a Flux KVS Namespace>`
+2. :ref:`Determine the managed directories for each application <Determine the Managed Directories for Each Application>`
+3. :ref:`Load DYAD's Flux module <Load DYAD's Flux Module>`
+4. :ref:`Configure and run the DYAD-enabled applications <Configure and Run the DYAD-Enabled Applications>`
+
+Create a Flux KVS Namespace
+***************************
+
+The first step in running DYAD is to create a namespace in the Flux's hierarchical key-value store (KVS).
```
Comment (JaeseungYeom) [inline docs/getting_started.rst:184]: Mention that this is to isolate DYAD from the rest of key-value entires used by Flux.

**Review by JaeseungYeom — APPROVED** [review]

File: docs/getting_started.rst (hunk 4 above)
Comment (ilumsden) [reply docs/getting_started.rst:201]: This is already added as a note at the end of this section. I chose to do this as a note to make it stand out to anyone reading the page.

Comment (JaeseungYeom) [reply docs/getting_started.rst:201]: I see. All good then.

File: docs/getting_started.rst [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+|                             |                 |              |         | within DYAD's namespace                                         |
++-----------------------------+-----------------+--------------+---------+-----------------------------------------------------------------+
+| :code:`DYAD_KEY_BINS`       | Integer         | No           | 1024    | The maximum number of unique values for the keys associated     |
+|                             |                 |              |         |                                                                 |
+|                             |                 |              |         | with any given level of Flux's hierarchical KVS within          |
+|                             |                 |              |         |                                                                 |
+|                             |                 |              |         | DYAD's namespace                                                |
++-----------------------------+-----------------+--------------+---------+-----------------------------------------------------------------+
+
```
Comment (JaeseungYeom) [inline docs/getting_started.rst:281]: A comment for the need/purpose of DYAD_KEY. It is for searching performance scalability.

**Pull request merged**